packages/block-editor/src/components/block-preview/index.js

```diff
@@ -63,3 +69,58 @@ export function BlockPreview( {
  * @return {WPComponent} The component to be rendered.
  */
 export default memo( BlockPreview );
+
+/**
+ * This hook is used to lightly mark an element as a block preview wrapper
+ * element. Call this hook and pass the returned props to the element to mark as
+ * a block preview wrapper, automatically rendering inner blocks as children. If
+ * you define a ref for the element, it is important to pass the ref to this
+ * hook, which the hook in turn will pass to the component through the props it
+ * returns. Optionally, you can also pass any other props through this hook, and
+ * they will be merged and returned.
+ *
+ * @param {Object}    options                      Preview options.
+ * @param {WPBlock[]} options.blocks               Block objects.
+ * @param {Object}    options.props                Optional. Props to pass to the element. Must contain
+ *                                                 the ref if one is defined.
+ * @param {Object}    options.__experimentalLayout Layout settings to be used in the preview.
+ *
+ */
+export function useBlockPreview( {
+	blocks,
+	props = {},
+	__experimentalLayout,
+} ) {
+	const originalSettings = useSelect(
+		( select ) => select( blockEditorStore ).getSettings(),
+		[]
+	);
+	const disabledRef = useDisabled();
+	const ref = useMergeRefs( [ props.ref, disabledRef ] );
+	const settings = useMemo( () => {
```

Comment:
Feel free to ignore if you think it's more readable as is. 

```suggestion
	() => ( { ...originalSettings, __experimentalBlockPatterns: [] } ),
```

Reply:
That does look more readable to me — I think from memory I'd just copied this bit from the other BlockPreview component, so the shorter syntax looks nice. I'll follow up.

Reply:
Pushed in https://github.com/WordPress/gutenberg/pull/36431/commits/a00533ecad7f0721af1103d62a6deb48320d1e87 👍